system/jlib/jsocket.cpp

```diff
@@ -5183,24 +5204,37 @@ class CSocketEpollHandler: implements ISocketSelectHandler, public CInterface
     void remove(ISocket *sock)
     {
         CriticalBlock block(sect);
-        unsigned nt = threads.ordinality();
-        ForEachItemIn(i,threads)
+        unsigned nt = threads.size();
+        bool removed = false;
+        std::list<Owned<CSocketEpollThread>>::iterator it = threads.begin();
+        while (it != threads.end())
         {
-            if (threads.item(i).remove(sock, nt))
-                break;
+            if (!removed)
+            {
+                if ((*it)->remove(sock, nt) && sock)
```

Comment:
can sock be null?

Reply:
If it was null then remove(sock) could return True so I was just being sure.